lib/core/types.hpp

```diff
@@ -1,7 +1,17 @@
 #pragma once
+#include <stdint.h>
 
 namespace hyped::core {
 
 enum class LowOrHigh { kLow = 0, kHigh };
 
+struct CanFrame
+{
+    uint32_t can_id; /* 32 bit CAN_ID + EFF/RTR/ERR flags */
```

Comment:
Are we certain that that all these numbers are strictly positive? I don't know CAN, just a quick sanity check.

Reply:
Don't think CAN ids can be negative

Reply:
The [socketCAN spec](https://docs.kernel.org/networking/can.html) defines them this way so I'm pretty sure it is true.
CAN frame IDs are essentially bitmasks so don't really have negatives, CAN frames can't have negative lengths, the data could be negative but this is representing the actual bytes we receive, rather than the specific values. Pad, res0, and res1 are for things like a checksum and are dealt with by socketCAN instead of us.

Reply:
Yeah i reckon this is fine

Reply:
Ok fair enough, just checking that we're not defaulting to `uint` without considering data values.